app/src/main/java/org/mozilla/fenix/compose/tabstray/TabListItem.kt

```diff
@@ -0,0 +1,185 @@
+fun TabListItem(
+    tab: TabSessionState,
+    isSelected: Boolean = false,
+    multiSelectionEnabled: Boolean = false,
+    multiSelectionSelected: Boolean = false,
+    onCloseClicked: ((TabSessionState) -> Unit)? = null,
+    onMediaClicked: ((TabSessionState) -> Unit)? = null,
+) {
+    val contentBackgroundColor = if (isSelected) {
+        R.color.tab_tray_item_selected_background_normal_theme
+    } else {
+        R.color.tab_tray_item_background_normal_theme
+    }
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(dimensionResource(id = R.dimen.tab_tray_tab_item_height))
+            .background(colorResource(id = contentBackgroundColor))
+    ) {
+
+        Thumbnail(
+            tab,
+            multiSelectionEnabled,
+            multiSelectionSelected,
+            Modifier.weight(2f),
+            onMediaClicked
+        )
+
+        val descriptionWeight = if (multiSelectionEnabled) 4f else 3f
+
+        DescriptionSection(
+            modifier = Modifier.weight(descriptionWeight),
+            tab = tab
+        )
+
+        CloseIcon(
+            tab = tab,
+            visible = !multiSelectionEnabled,
+            modifier = Modifier.weight(1f),
+            onCloseClicked = onCloseClicked
+        )
+    }
+}
+
+@Composable
+private fun Thumbnail(
+    tab: TabSessionState,
+    multiSelectionEnabled: Boolean,
+    isSelected: Boolean,
+    modifier: Modifier,
+    onMediaIconClicked: ((TabSessionState) -> Unit)?
+) {
+    Box(
+        modifier = modifier
+            .padding(start = 16.dp, end = 12.dp, top = 8.dp, bottom = 8.dp)
+    ) {
+        TabThumbnail(
+            tab = tab,
+            modifier = Modifier.size(92.dp, 72.dp),
+            thumbnailDescription = stringResource(id = R.string.mozac_browser_tabstray_open_tab),
+            contentScale = ContentScale.FillWidth,
+            alignment = Alignment.TopCenter
+        )
+
+        if (isSelected) {
+            Card(
+                modifier = Modifier
+                    .size(40.dp)
+                    .align(Alignment.Center),
+                shape = CircleShape,
+                backgroundColor = colorResource(id = R.color.accent_normal_theme),
+            ) {
+                Icon(
+                    painter = painterResource(id = R.drawable.mozac_ic_check),
+                    modifier = Modifier
+                        .matchParentSize()
+                        .padding(8.dp),
+                    contentDescription = null,
+                    tint = colorResource(id = R.color.mozac_ui_icons_fill)
+                )
+            }
+        }
+        if (!multiSelectionEnabled) {
+            MediaImage(tab, onMediaIconClicked, Modifier.align(Alignment.Center))
+        }
+    }
+}
+
+@Composable
+private fun CloseIcon(
+    tab: TabSessionState,
+    visible: Boolean,
+    modifier: Modifier,
+    onCloseClicked: ((TabSessionState) -> Unit)?
+) {
+    if (visible) {
```

Comment:
We typically don't want to execute a composable function if it can be avoid. I would move the `if (visible)` check outside of `CloseIcon` and let the caller decide whether or not `CloseIcon` needs to be called.